chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml

```diff
@@ -0,0 +1,133 @@
+
+- name: Get ephemeral-storage Request storage unit
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ a_ns }} -o jsonpath='{.spec.containers[?(@.name=="{{ c_container }}")].resources.requests.ephemeral-storage}' | grep -Eo '[[:alpha:]]+'
+  args:
+    executable: /bin/bash
+  register: req_value_storage
+
+
+- name: Fetch the value of Request ephemeral-storage Unit in KB
+  set_fact:
+    req_storage_KB: "{{ req_value_storage.stdout }}"
+
+- name: Fetch the value of Limit ephemeral-storage Unit in KB
+  set_fact:
+    limit_storage_KB: "{{ limit_value_storage.stdout }}"
+
+- include_tasks: /chaoslib/litmus/disk_fill/convert_fill_percentage.yml
+
+- name: Get the Container ID
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ a_ns }} -o jsonpath='{.status.containerStatuses[?(@.name=="{{ c_container }}")].containerID}' | awk -F'//' '{print $2}'
+  args:
+    executable: /bin/bash
+  register: containerID
+
+- name: Identify the application node
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ a_ns }} --no-headers -o custom-columns=:spec.nodeName
+  args:
+    executable: /bin/bash
+  register: app_node
+
+- name: Create the DaemonSet for Disk-Fill
+  shell: >
+    kubectl apply -f /chaoslib/litmus/disk_fill/disk_fill_ds.yaml -n {{ a_ns }}
+  args:
+    executable: /bin/bash
+  
+- name: Confirm that the disk-fill daemonSet is running on all nodes
+  shell: >
+    kubectl get pods -l app=disk-fill --no-headers --template "{{range .items}}{{if eq .spec.nodeName $(kubectl get nodes --no-headers --field-selector spec.unschedulable=false | grep Ready | awk '{printf "\"%s\" ", $1}')}}{{.status.phase}}{{\"\n\"}}{{end}}{{end}}" -n {{ a_ns }} | sort | uniq
+  args:
+    executable: /bin/bash
+  register: result
+  until: "result.stdout == 'Running'"
+  delay: 1
+  retries: 60
+  ignore_errors: true
+  
+- name: Record the disk-fill pod on app node
+  shell: >
+    kubectl get pods -l app=disk-fill -o jsonpath='{.items[?(@.spec.nodeName=="{{app_node.stdout}}")].metadata.name}'
+  args:
+    executable: /bin/bash
+  register: disk_fill_pod
+
+- name: Exec into the pod to get ephemeral-storage size
+  shell: >
+    kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ a_ns }} -- sh -c "cd /diskfill && cd {{ containerID.stdout }} && du -k |tail -n1 | awk '{print $1}'"
+  args:
+    executable: /bin/bash
+  register: pod_current_size_full
+
+- name: Echo the filled storage in KB to get the integer part for multiplication
+  shell: >
+    echo {{ pod_current_size_full.stdout }} | grep -Eo '+|[0-9]+'
+  args:
+    executable: /bin/bash
+  register: pod_current_size
+
+- name: Total Used Ephemeral Storage in KB
+  set_fact:
+    used_storage: "{{ pod_current_size.stdout }}"
+  
+- name: Subtract to get the file allocation size
+  set_fact:
+    fill_storage: "{{ to_fill|int - used_storage|int }}"
+
+- include_tasks: /chaoslib/litmus/disk_fill/file_creation.yml
+
+- include_tasks: /utils/common/status_app_pod.yml
```

Comment:
can you please add the name field -easy for debug

Reply:
@shubhamchaudhary not needed, all tasks name are self explainatory